### PR TITLE
fix(sca): Set prisma token and scan packages by v2 for IDE scans

### PR DESCRIPTION
**Pull request by matansha**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sca_package_2/runner.py [hunk 1]
```diff
@@ -45,8 +46,16 @@ def prepare_and_scan(
             logging.info("The --bc-api-key flag needs to be set to run SCA package scanning")
             return None
 
+        if bc_integration.bc_source and bc_integration.bc_source.name in [BCSourceType.VSCODE, BCSourceType.JETBRAINS] \
```
Comment (gruebel) [inline checkov/sca_package_2/runner.py:49]: can you create a set in `checkov.common.bridgecrew.bc_source` with both values and reference it here, then we can directly extend there, if we support more someday and we can leverage it in other areas, if needed 🙂

File: checkov/sca_package_2/runner.py [hunk 2]
```diff
@@ -45,8 +46,16 @@ def prepare_and_scan(
...
+                and not bc_integration.is_prisma_integration():
+            logging.info("The --bc-api-key flag needs to be set to Prisma token for SCA scan for vscode or jetbrains extention")
```
Comment (gruebel) [inline checkov/sca_package_2/runner.py:51]:
```suggestion
            logging.info("The --bc-api-key flag needs to be set to a Bridgecrew or Prisma token for SCA scan for vscode or jetbrains extention")
```

Comment (gruebel) [reply checkov/sca_package_2/runner.py:51]: @tsmithv11 can you help with some proper wording here 😄

**Review by gruebel — COMMENTED** [review]
nice work, looks good to me

File: checkov/sca_package_2/runner.py (hunk 2 above)
Comment (matansha) [reply checkov/sca_package_2/runner.py:51]: We need to block Bridgecrew and allow only Prisma

File: checkov/sca_package_2/runner.py (hunk 1 above)
Comment (matansha) [reply checkov/sca_package_2/runner.py]: Fixed

**Review by gruebel — APPROVED** [review]
🍪

**Review by NoaAzoulay — APPROVED** [review]

File: checkov/main.py [hunk 3]
```diff
@@ -683,7 +683,7 @@ def run(self, banner: str = checkov_banner) -> int | None:
                 logger.debug('Using --list; setting source to DISABLED')
                 source = SourceTypes[BCSourceType.DISABLED]
 
-            if CHECKOV_RUN_SCA_PACKAGE_SCAN_V2 and source.upload_results:
+            if CHECKOV_RUN_SCA_PACKAGE_SCAN_V2:
                 self.runners.append(sca_package_runner_2())
             else:
                 self.runners.append(sca_package_runner())
```
Comment (ayajbara) [inline checkov/main.py:689]: also here I think we dont need this anymore, and need to add the runner 2 to runner list `DEFAULT_RUNNERS`

**Pull request merged**